### PR TITLE
LPS-122876 [BUG] Fix problems with the redirection from the Web Content Editor to Content Dashboard

### PR DESCRIPTION
**Pull request by cgoncas**
**Motivation**

The filters applied to the Content Dashboard disappears when you come back to the Web Content Editor after publishing.

**Proposed Solution**

Use the right redirect when the Edit URL is being generated.

**How to test it**

1. Create a Web Content and add one category (category1)
2. Go to the Content Dashboard
3. Filter by the previous category (category1)
4. Click on the Web Content pencil to go to the edition
5. Publish without any change

Integration tests have been included within this pull request

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
Wrong LPS, this pull continues in https://github.com/liferay-tango/liferay-portal/pull/547

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci: forward

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:forward

File: modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/EditJournalArticleContentDashboardItemAction.java [hunk 1]
```diff
@@ -66,10 +74,20 @@ public Type getType() {
 
 	@Override
 	public String getURL() {
+		Portlet portlet = _portletLocalService.getPortletById(
+			_portal.getPortletId(_httpServletRequest));
+
+		HttpServletRequest httpServletRequest =
```
Comment (darquesdev) [inline modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/EditJournalArticleContentDashboardItemAction.java:80]: Hi @cgoncas! why do we need to create a new request here? in which situation should I use this method? thx

Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/EditJournalArticleContentDashboardItemAction.java:80]: Hey @darquesdev !!

As a rule of thumb, you should never use it :)

image

If you are curious about why do we need to do it here, please, take a look at how the backlink is generated in the Web Content Editor.

I think this is a really good exercise of how the portal sometimes works, so please, add it to your Liferay learning plan. We can talk about it in our 1:1.

Thanks!!!

Comment (darquesdev) [reply modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/EditJournalArticleContentDashboardItemAction.java:80]: hahaha, Ok, added to my learning plan, thanks!

**Pull request closed without merging**